### ISSUE TITLE
Translation, option to add translations to keys

### ISSUE THREAD

**Issue by Nils-1st-retail** [php-translation/symfony-bundle#179]
Hi, 

I have been working on creating an extractor for routes defined in annotations. (Migrating from JMSI18nRoutingBundle with JMSTranslationBundle) to using this bundle for the translation part.
I'm missing an option to specify a value for the translation from the extractor.

My suggestion would be to either add an extra optional parameter to SourceLocation defaulting to null or allow to set the translation in the context array. 

Then in the Symfony Importer service change line 118 from
`$catalogue->set($key, null, $domain);`
to 
`$catalogue->set($key, $sourceLocation->getTranslation(), $domain);`
or
`$catalogue->set($key, isset($context["translation"]) ? $context["translation"] : null, $domain);`

The problem atm. is that because all the route values are set as null, all my routes defaults to '/' until I provide a translation.

**Comment by Nyholm** [comment]
Hm, do you mind sharing some code and you use case more properly?
It sounds like something we definitely would like to support but I think the approach is wrong. 

Since one never would store translation in the source code, the `SourceLocation` would never have a translation.

**Comment by Nils-1st-retail** [comment]
The `JMSI18nRoutingBundle`, as default looks for a route translation under the "routes" domain.
The key of the translation is the full path of the route.

I have added a visitor extending the `basePhpVisitor` to the `PHPFileExtractor` that parses all my classes locating the routes. 

From the Visitor the only (current) option is to make a call to the `addLocation` function which creates a new `SourceLocation`

atm. I added a hack where I in the `Visitor` add the found locations like so.
`$this->addLocation($name, $node->getLine(), null, ['domain' => 'routes','translation'=>$this->prefix.$route]);`

and then in the `Translation\Bundle\Service\Importer`, reads the translation from the `SourceLocation`'s context, too add the default translation.

I agree that the translation would never belong in the source code, but then it would be nice to have another way of piping translations from a `Visitor` to the catalogue.
